### ISSUE TITLE
--calcite-color-status-success color doesn't have enough contrast

### ISSUE THREAD

**Issue by HauTruong146** [Esri/calcite-design-system#12158]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

Issue: https://github.com/ArcGIS/business-analyst-webapp/issues/20734

The color contrast of ```--calcite-color-status-success``` against white foreground is not sufficient. 

### Actual Behavior

There are graphical objects with a color contrast ratio below 3.00:1.

Foreground color: #FFFFFF
Background color: #35AC46
Contrast ratio: 2.9:1

### Expected Behavior

Ensure parts of graphical objects essential for understanding content have sufficient contrast. The required minimum contrast ratio is 3.00:1.

Common examples of qualifying objects include lines in a chart, meaningful icons, and annotations within an image. Graphics that require particular presentation to preserve their meaning are exempt from this requirement.

To calculate color contrast ratios, use a tool such as the Level Access Accessible Color Picker Chrome extension: https://chrome.google.com/webstore/detail/accessible-color-picker/bgfhbflmeekopanooidljpnmnljdihld or the Color Contrast Checker: https://www.levelaccess.com/color-contrast-checker-new/

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/foundations/colors/

### Reproduction Steps

1. Locate any success icon that corresponds to a successfully completed tour in the 'Guided tours' dialog.
2. Use any contrast measuring tool like https://www.levelaccess.com/color-contrast-checker-new/ and measure the contrast ratio between the icon and its background color.

### Reproduction Version

3.0.3

### Relevant Info

<img width="2056" alt="Image" src="https://github.com/user-attachments/assets/ce1aa3f1-2dea-4d2e-bb92-ae101be0585b" />

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by geospatialem** [comment]
Color should not be used alone to define content, and while Calcite provides color and theming, including `--calcite-color-status-success color`, it is up to the developer to leverage other means to communicate information other than color. More information can be found in the [Calcite accessibility color](https://developers.arcgis.com/calcite-design-system/foundations/accessibility/#color) section, where it is recommended to use a combination of color, shapes, patterns, and text to convey meaning and providing alternatives for complex maps, charts, and graphics.

@HauTruong146 Can you provide more information to us in the form of a code snippet? The screenshot above doesn't indicate where the color is being used, but if its to indicate whether something is "checked off" or not, it appears their is distinction between the two states already in the application.

**Comment by HauTruong146** [comment]
Hi @geospatialem , I included code snippets below. We use the icon as a "check mark" along with the "--calcite-color-status-success" to indicate to the user that they have completed that guided tour. We also provided aria-label with the calcite-icon as "Completed tour".

- ### Where we create the calcite-icon:
```
const tourIcon = createSmallIcon(
  {
    icon: item.completed ? "check-circle-f" : "circle",
    size: 24,
    kind: item.completed ? "success" : "neutral",
    // ariaHidden: "false",
    // ariaLabel: item.completed ? nls.completedTourIconLabel : nls.notCompletedTourIconLabel
  },
  rootInner
);
```

- ### Where we use the kindClass as a class name to assign css styling to.
```
export function createSmallIcon(params: SimpleIconParams, container?: HTMLElement, position?: PlacePosition): Icon {
  let { class: cls, kind, size, ...rest } = params;
  let kindClass: string;
  switch (kind) {
    case "danger":
      kindClass = "dangerColor";
      break;
    case "warning":
      kindClass = "warningColor";
      break;
    case "success":
      kindClass = "successColor";
      break;
  }
  if (kindClass) {
    cls = kindClass + (cls ? " " + cls : "");
  }

  const icon = new Icon({ scale: "s", class: cls, size: size === "auto" ? null : size, ...rest });
  if (size === "auto") {
    const node = icon.domNode;
    node.style.blockSize = "";
    node.style.inlineSize = "";
    node.style.minInlineSize = "";
    node.style.minBlockSize = "";
  }
  container && icon.placeAt(container, position);
  return icon;
}
```

- ### In the .scss file, we set color to element using class name with different calcite colors
```
.esriGECalcite {
    &.dangerColor {
      color: var(--calcite-color-status-danger);
    }
    &.warningColor {
      color: var(--calcite-color-status-warning);
    }
    &.successColor {
      color: var(--calcite-color-status-success);
    }
  }
```

**Comment by geospatialem** [comment]
> I included code snippets below. We use the icon as a "check mark" along with the "--calcite-color-status-success" to indicate to the user that they have completed that guided tour. We also provided aria-label with the calcite-icon as "Completed tour".

Thanks for the quick reply! 🏆  That should fit the use case mentioned [above](https://github.com/Esri/calcite-design-system/issues/12158#issuecomment-2885006408), where color alone is not being depicted, and there are different states **in addition to color**.

The above doesn't seem like a bug or valid, since color, context, and a check mark are displayed for the context. @HauTruong146 if you agree, think the app fits the Success Criterion of [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) and we can close up the above as unplanned.

**Comment by geospatialem** [comment]
Closing the above out as unplanned, refer to https://github.com/Esri/calcite-design-system/issues/12158#issuecomment-2885075426 and https://github.com/Esri/calcite-design-system/issues/12158#issuecomment-2885085424.
